clang/lib/Tooling/DependencyScanning/DependencyScanningFilesystem.cpp

```diff
@@ -270,6 +270,12 @@ DependencyScanningWorkerFilesystem::status(const Twine &Path) {
   return Result->getStatus();
 }
 
+bool
+DependencyScanningWorkerFilesystem::exists(const Twine &Path) {
+  llvm::ErrorOr<llvm::vfs::Status> Status = status(Path);
```

Comment:
Would be good to explain why it's necessary & okay to kill the `exists()` optimization here.

Reply:
Which optimization are you referring to?

Reply:
The one from Ben's commit: using `exists()` instead of `status()` can be implemented more efficiently on some VFSs. It can be surprising that `DependencyScanningWorkerFilesystem` just throws that out of the window. But it's typically wrapped around `RealFileSystem` that doesn't specialize `exists()`, and we could not cache the result of just `exists()` here anyway.